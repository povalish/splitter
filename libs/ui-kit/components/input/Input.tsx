import React, { useId } from 'react';

import { IWebInput } from './Input.interface';
import { container, label as labelStyles, input } from './Input.classes';

//
//

export const Input: React.FC<IWebInput> = ({ className, value, label, ...inputProps }) => {
  const id = useId();

  return (
    <div className={container({ className })}>
      <label htmlFor={id} className={labelStyles({ active: !!value })}>
        {label}
      </label>
      <input id={id} className={input()} type='text' value={value} {...inputProps} />
    </div>
  );
};
