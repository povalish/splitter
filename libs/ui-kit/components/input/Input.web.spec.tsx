import React from 'react';
import { render } from '@testing-library/react';
import { Input } from './Input';

//
//

describe('Input component', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <Input label='Text input' type='text' className='min-w-[300px]' />,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
