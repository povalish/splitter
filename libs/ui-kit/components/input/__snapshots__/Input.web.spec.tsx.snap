// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render successfully 1`] = `
<body>
  <div>
    <div
      class="relative pt-3 group min-w-[300px]"
    >
      <label
        class="
  absolute 
  text-colorText whitespace-nowrap 

  group-focus-within:text-colorAccent
  group-focus-within:top-[-5px] group-focus-within:left-0 
  group-focus-within:text-sm group-focus-within:font-normal

  transition-all 
  pointer-events-none
 top-[20px] left-3 text-base font-bold"
        for=":r0:"
      >
        Text input
      </label>
      <input
        class="
  w-full
  bg-transparent px-3 py-2 
  border-b-[3px] border-b-colorText
  text-colorText text-base font-bold

  disabled:bg-colorDisabled disabled:border-colorDisabled disabled:scale-100 disabled:cursor-not-allowed
  focus:border-b-[3px] focus:border-colorAccent focus:outline-none

  transition-all
"
        id=":r0:"
        type="text"
        value=""
      />
    </div>
  </div>
</body>
`;
